content/posts/2019/05/first-contribution.eng.md

```diff
@@ -0,0 +1,70 @@
+ 
+C# Development Environment 
+------------
+
+As mentioned earlier, I used to mainly develop in Python on Mac. I often contributed to Python open source projects that interested me and this was easy because most Python developers also developed on Mac and were familiar with the open source environment. But as I was about to build a C# development environment on Mac, I found myself stuck from the very first step, unsure about what to install. Fortunately, there were projects like [Mono][] and [.NET Core][2] that enabled .NET development on Mac, and editors such as [VS Code][] that also supported plug-ins for C# development. So I was able to set up an environment without much difficulty. Now that Libplanet also provides a [developer guide][3] and a [environment setup guide][4], I think it will be much easier for developers who want to contribute to the project.
+
+[Mono]: https://www.mono-project.com/
+[2]: https://en.wikipedia.org/wiki/.NET_Core
+[VS Code]: https://code.visualstudio.com/
+[3]: https://github.com/planetarium/libplanet/blob/master/CONTRIBUTING.md
+[4]: https://gist.github.com/dahlia/5333634f62509293cd46c0e4ba65b2f5
+
+What to Contribute?
+------------------
+
+Libplanet’s address uses [Ethereum][5]'s address format, which uses a checksum in a combination of upper and lower case letters proposed in [EIP-55][]. For example:
+
+Hexadecimal
+:  `0xd1220a0cf47c7b9be7a2e6ba89f429762e7b9adb`
+
+EIP-55 Checksum
+:  `0xD1220A0cf47c7B9Be7A2E6BA89F429762e7b9aDb`
+
+
+When I first contributed to Libplanet, the project used a hexadecimal consisting of only lowercase letters for its address format. Therefore, this issue had to be implemented so that the address string representation was expressed in the EIP-55 checksum format used by Ethereum. For more information, see the corresponding [issue][6] and [pull request][7].
+
+Although I had little development experience and background in C#, it was a simple issue that did not require many modules to be fixed, so I was able to make my first contribution rather easily.
+
+[5]: https://www.ethereum.org/
+[EIP-55]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+[6]: https://github.com/planetarium/libplanet/issues/33
+[7]: https://github.com/planetarium/libplanet/pull/43
+
+## What I Learned Through My First Contribution
+
+What I found out from my first contribution was that the C# development environment on Mac was more easy than I had anticipated. The language itself was not difficult because C# was oriented towards a popularized multi-paradigm programming language. While the .NET development environment on Mac was not too difficult to develop, support for various tools on third-party libraries and non-window platforms was still insufficient. However, this is expected to improve gradually thanks to Microsoft's recent open source and multi-platform policies. 
```

Comment:
```suggestion
What I found out from my first contribution was that the C# development environment on Mac was more easy than I had anticipated. The language itself was not difficult because C# was oriented towards a popularized multi-paradigm programming language. While the .NET development environment on Mac was not too difficult to develop, support for various tools on third-party libraries and non-window platforms was still insufficient. However, this is expected to improve gradually thanks to Microsoft's recent open source and multi-platform policies. 

```